Sources/MockoloFramework/Models/Import.swift

```diff
@@ -0,0 +1,178 @@
+    }
+    
+    init(
+        moduleName: String,
+        modifier: Modifier? = nil,
+        prefix: String? = nil,
+        suffix: String? = nil
+    ) {
+        self.moduleName = moduleName
+        self.modifier = modifier
+        self.prefix = prefix
+        self.suffix = suffix
+    }
+}
+ 
+extension Import {
+    
+    /// Returns a copy with a `.testable` modifier
+    var asTestable: Import {
+        var new = self
+        new.modifier = .testable
+        return new
+    }
+    
+    /// Creates an `Import` by parsing a `String` provided by `Generator`.
+    /// It is typically a single line, but can be wrapped by `#if FOO\n...\n#endif` when it's a nested macro.
+    init?(line: String) {
+        guard let importSpaceRange = line.range(of: String.importSpace) else { return nil }
+        
+        let firstNewlineIndex = line.firstIndex(of: "\n")
+        let lastNewlineIndex = line.lastIndex(of: "\n")
+        let startIndex = firstNewlineIndex ?? line.startIndex
+        let endIndex = lastNewlineIndex ?? line.endIndex
+        
+        moduleName = String(line[importSpaceRange.upperBound..<endIndex])
+        
+        if importSpaceRange.lowerBound == startIndex {
+            modifier = nil
+        } else {
+            let modifierEndIndex = line.index(before: importSpaceRange.lowerBound)
+            modifier = Modifier(rawValue: String(line[startIndex..<modifierEndIndex]))
+        }
+        
+        `prefix` = firstNewlineIndex.map { String(line[...$0]) }
+        suffix = lastNewlineIndex.map { String(line[$0...]) }
+    }
+}
+
+extension Array where Element == Import {
+    
+    /// Prepares a list of imports for output:
+    /// - consolidates imports of the same module
+    /// - maintains the highest given ACL for a given module
+    /// - overrides ACL with `@testable` when present
+    /// - sorts by module name
+    func resolved() -> [Import] {
+        var modifierByModuleName = [String: Import.Modifier]()
+        var prefixByModuleName = [String: String]()
+        var suffixByModuleName = [String: String]()
+        
+        for imp in self {
+            switch (imp.modifier, modifierByModuleName[imp.moduleName]) {
+            case let (.acl(acl), .acl(existingACL)):
+                if acl.rank > existingACL.rank {
+                    modifierByModuleName[imp.moduleName] = .acl(acl)
+                }
+            case (.testable, .acl(.internal)), (.testable, .acl(.package)):
+                modifierByModuleName[imp.moduleName] = .testable
+            case (.some(let modifier), .none):
+                modifierByModuleName[imp.moduleName] = modifier
+            default:
+                break
+            }
+            
+            if let prefix = imp.prefix {
+                prefixByModuleName[imp.moduleName] = prefix
+            }
+            if let suffix = imp.suffix {
+                suffixByModuleName[imp.moduleName] = suffix
+            }
```

Comment:
I don't think this logic is correct(Should write separate `#if` imports even if it's redundant), but since the original implementation was also incorrect, passing the tests should be sufficient.